kinesis/aggregators.py

```diff
@@ -172,3 +179,44 @@ def parse(self, data):
             i += header_offset + size + 2
             if i == length:
                 break
+
+
+class KPLAggregator(Aggregator):
+    """
+    KPL Aggregated Record Aggregation
+    See: https://github.com/awslabs/kinesis-aggregation/tree/master/python
+    """
+
+    def __init__(self, max_size=None):
+        if max_size is not None:
+            # Ideally, aws_kinesis_agg will make this configurable at some point..
+            log.warning("KPL Aggregation does not support max_size. Using aws_kinesis_agg.MAX_BYTES_PER_RECORD of {}").format(aws_kinesis_agg.MAX_BYTES_PER_RECORD)
```

Comment:
I thought about copying much of the size-checks that aws_kinesis_agg employs instead, or potentially monkey patching the constant, but I can try to see if awslabs would make this configurable upstream, which would make this much simpler.

Reply:
```
# Kinesis Limits
# (https://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecord.html)
MAX_BYTES_PER_RECORD = 1024*1024  # 1 MB
```

right, the limit for the stream. I guess they will wonder why you would ever want to set a lower limit? hehe

Can't hurt to ask I guess. I would say maybe raise exception in this case rather than warning. It's probably fine as is.

Reply:
Raising exception makes sense, similar to the ValidationErrors in the BaseAggregator. I *think* that multi-lang-daemon service lets you configure this as a setting. If that's so, I think there's a good argument for making it configurable in the awslabs project. I'll try to find that option and open an issue on the awslabs project to see if anyone has an opinion there.

Reply:
Created https://github.com/awslabs/kinesis-aggregation/pull/129

Reply:
@hampsterx I switched to raising the ValidationError instead of logging a warning and added a test for that assertion. I will follow up here if the change is merged into the awslabs project, but I don't think it's worth blocking this PR to wait for that.
I also tried running this with a sample producer+consumer for both the KPLJsonProcessor and KPLStringProcessor with a real kinesis stream and was able to read the data I wrote, so that sanity check is complete. Please merge if you accept the changes, or let me know what other changes I can make. Thanks!